packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -229,14 +299,61 @@ class Context {
         nativeAssetsPath,
```

Comment:
I wonder why rsync'ing would ever want delete to be false and keep around stale files, it looks intentional from https://github.com/flutter/flutter/pull/130494 but sounds like a bug.

Reply:
@dcharkes Do you remember if this was done intentionally?

Reply:
No, I don't think it's intentional to leave stale files. I think it's unlikely that we'd have three builds where we have the same native asset in the first and the third build.

Reply:
Okay I figured out why this was. It was `rsync`ing the entire `nativeAssetsDir` directory (which contained `.framework` sub directories) into the `Frameworks` directory. So if `--delete` was used, it would delete any frameworks that were already in the `Frameworks` directory. I have changed it so it `rsync`s each individual `.framework` from the `nativeAssetsDir`